File: 001.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    int length = nums.size();
    for (int i = 0; i < length-1; i++) {
        for (int j = i+1; j < length; j++) {
            if (nums[i] + nums[j] == target) {
                vector<int> result;
                result.push_back(i);
                result.push_back(j);
                return result;
            }
        }
    }
}

int main() {
	int nums[4] = {2, 7, 11, 15};
	vector<int> v(nums, nums+4);
	twoSum(v, 9);
}

File: Array/001.py
def twoSum(self, nums, target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: List[int]
    """
    s = {}
    for i in range(len(nums)):
        if s.has_key(nums[i]):
            # 不能重复取同一个元素
            if s[nums[i]] != i:
                return [i, s[nums[i]]]
        else:
            # 记录对应位置
            s[target - nums[i]] = i
